### PR TITLE
Updating EntitySave action to use doExecute()

### PR DESCRIPTION
**Pull request by ndeet**
See d.o issue: https://www.drupal.org/node/2572119

### CODE REVIEW

File: src/Plugin/RulesAction/EntitySave.php [hunk 1]
```diff
@@ -41,13 +42,16 @@ class EntitySave extends RulesActionBase {
...
+   * @param EntityInterface
+   *    The entity to be saved.
    */
-  public function execute() {
+  public function doExecute(EntityInterface $entity) {
     // We only need to do something here if the immediate flag is set, otherwise
     // the entity will be auto-saved after the execution.
     if ((bool) $this->getContextValue('immediate')) {
-      $entity = $this->getContextValue('entity');
+      //$entity = $this->getContextValue('entity');
```
Comment (dasjo) [inline src/Plugin/RulesAction/EntitySave.php]: Let's really remove that one :)

**Comment by ndeet** [comment]
removed commented line

File: src/Plugin/RulesAction/EntitySave.php [hunk 2]
```diff
@@ -41,13 +42,15 @@ class EntitySave extends RulesActionBase {
   protected $saveLater = TRUE;
 
   /**
-   * {@inheritdoc}
+   * Saves the Entity.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *    The entity to be saved.
```
Comment (klausi) [inline src/Plugin/RulesAction/EntitySave.php]: should use 2 space indentation

**Comment by ndeet** [comment]
Fixed.

File: src/Plugin/RulesAction/EntitySave.php [hunk 3]
```diff
@@ -41,13 +42,15 @@ class EntitySave extends RulesActionBase {
...
 
   /**
-   * {@inheritdoc}
+   * Saves the Entity.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity to be saved.
    */
-  public function execute() {
+  public function doExecute(EntityInterface $entity) {
```
Comment (klausi) [inline src/Plugin/RulesAction/EntitySave.php]: The visibility should be protected here

**Comment by ndeet** [comment]
Fixed.

File: src/Plugin/RulesAction/EntitySave.php [hunk 4]
```diff
@@ -41,13 +42,15 @@ class EntitySave extends RulesActionBase {
...
+   * Saves the Entity.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity to be saved.
    */
-  public function execute() {
+  protected function doExecute(EntityInterface $entity) {
     // We only need to do something here if the immediate flag is set, otherwise
     // the entity will be auto-saved after the execution.
     if ((bool) $this->getContextValue('immediate')) {
```
Comment (klausi) [inline src/Plugin/RulesAction/EntitySave.php]: The "immediate" context should also come in as method parameter, same as the entity.

**Comment by ndeet** [comment]
Right, missed that one. Fixed.

**Comment by klausi** [comment]
merged, thanks!

**Pull request closed without merging**